src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/update/actions/ValidateYFlowAction.java

```diff
@@ -145,12 +145,20 @@ protected Optional<Message> performWithResponse(State from, State to, Event even
 
     private String getMainAffinityFlowId(YFlow yFlow) {
         // TODO: maybe we should add filtering of one switch sub flows into method getSubFlows()
-        YSubFlow multiSwitchFlows = yFlow.getSubFlows().stream()
+        YSubFlow flowInAffinityGroup = yFlow.getSubFlows().stream()
+                .filter(sub -> sub.getFlow().getAffinityGroupId() != null)
+                .findAny()
+                .orElse(null);
+        if (flowInAffinityGroup != null) {
+            return flowInAffinityGroup.getFlow().getAffinityGroupId();
+        }
```

Comment:
In this particular case (just two stream elements) I'd do something like 
```
String affinityGroup = yFlow.getSubFlows().stream()
                .mapToString(sub -> sub.getFlow().getAffinityGroupId())
                .filter(affinityGroupId -> affinityGroupId != null)
                .findAny()  //can be replaced by .reduce()
                .collect(joining())
if (affinityGroup != "") {
                    return affinityGroup}
```
This way you call .getFlow().getAffinityGroupId() two times for each validate method call instead of three times depending on success.
Overall, I'd rather see method `String getAffinityGroupId()` in YFlow class.